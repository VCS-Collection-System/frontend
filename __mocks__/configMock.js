export const getConfig = (URL, configUrl, config) => {
  if(URL.indexOf(configUrl) > -1 ) {
    const country = URL.substring(configUrl.length);
    const countryConfig = config[country] || null;
    return Promise.resolve({data: countryConfig});
  } else {
    return Promise.reject({data: null});
  }
}