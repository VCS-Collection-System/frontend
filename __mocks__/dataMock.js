export const employeeInfo = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth":"2021-10-07",
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const attachmentObject = {
  "originalFileName":"blob",
  "size":100,
  "contentType":"image/png",
  "s3BucketName":"mockBucket",
  "s3UUID":"mockUUID"
}

export const vaxHistory = {
  "id":0,
  "employee": employeeInfo,
  "submissionDate":"2021-10-04T15:12:14.875",
  "review":{
     "reviewerEmployeeId":"z2",
     "reviewerNotes":"",
     "reviewDate":"2021-10-05T14:04:29.296",
     "reviewOutcome":"ACCEPTED",
     "rejectReason":""
  },
  "submittedBy":null,
  "autoApproved":false,
  "lastUpdatedTime":null,
  "lastUpdatedBy":null,
  "vaccineBrand":"MODERNA",
  "vaccineAdministrationDate":"2021-10-04",
  "fullVaccinatedFlag":false,
  "vaccineShotNumber":2,
  "documentType":"VaccineDocument",
  "attachment": {}
}

const taskEmployeeObject = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth": {
    "year": "2021",
    "monthValue": "12",
    "dayOfMonth": "16"
  },
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const mockTask = {
  "task-input-data": {
    "documentList": {
      "com.redhat.vcs.model.VaccineDocumentList": {
        "documents": [ {
          "com.redhat.vcs.model.VaccineCardDocument": {
            "employee": {
              "com.redhat.vcs.model.Employee": taskEmployeeObject
            },
            "vaccineBrand": {
              "com.redhat.vcs.model.VaccineBrand": "MODERNA"
            },
            "attachment": {
              "com.redhat.vcs.model.Attachment": attachmentObject
            },
            "vaccineAdministrationDate": {
              "year": "2021",
              "monthValue": "12",
              "dayOfMonth": "16"
            },
            "lotNumber": "mockLot"
          } }
        }
      }
    }
  }
}
