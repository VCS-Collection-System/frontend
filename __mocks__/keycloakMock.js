module.exports = {
    token: "bogustoken", 
    authenticated: false,
    loadUserProfile: jest.fn( ()=> {return Promise.resolve({
        attributes: {
            workforceid: ["mockId"]
        }
    })} ),
    login: jest.fn(),
    logout: jest.fn(),
    hasRealmRole: jest.fn(),
    profile: {
        firstName: "mockFirst",
        lastName: "mockLast",
        email: "mock@email.com",
        attributes: {
            workforceid: ["mockId"],
            agency: ["USA"]
        }
    }
};
