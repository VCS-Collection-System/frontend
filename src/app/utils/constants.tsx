declare global {  
  interface Window {
    REACT_APP_PAM_URL_BASE: string;
    REACT_APP_PAM_CONTAINER_ID: string;
    REACT_APP_KEYCLOAK_COUNTRY: string;
    REACT_APP_KEYCLOAK_EMAIL: string;
    REACT_APP_KEYCLOAK_FIRST_NAME: string;
    REACT_APP_KEYCLOAK_ID: string;
    REACT_APP_KEYCLOAK_LAST_NAME: string;
    REACT_APP_MAINTENANCE_BANNER: boolean;
    REACT_APP_MAINTENANCE_BANNER_TEXT: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_FEATURE_FLAGS: Array<Record<string, any>>;
    REACT_APP_MAX_DOSES: Record<string, number>;
    REACT_APP_CONSENT_LABEL: string;
    REACT_APP_ALTERNATE_CONSENT_LABEL: string;
    REACT_APP_BLOCKED_COUNTRIES: Array<string>;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_GLOBAL_CONFIG: Record<string, any>;
  }
}

// PAM urls
export const PAM_BASE_URL = window.REACT_APP_PAM_URL_BASE; 
export const SUBMIT_VAX_URL = PAM_BASE_URL + '/attestation/vax';
export const SUBMIT_TEST_URL = PAM_BASE_URL + '/attestation/covid-test-result';
export const PAM_TASK_URL = PAM_BASE_URL + '/tasks/';
export const RESERVED_TASK_URL = PAM_TASK_URL + 'owners?status=Reserved&user=';
export const INPROGRESS_TASK_URL = PAM_TASK_URL + 'owners?status=InProgress&user=';
export const ALL_TASK_URL = PAM_TASK_URL + 'owners?status=Ready&pageSize=200';
export const PRIORITY_URL = PAM_BASE_URL + '/query/priority-results';
export const PRIORITY_CONFIRM_URL = PAM_BASE_URL + '/query/priority-result/';
export const ATTACHMENT_URL = PAM_BASE_URL + '/attestation/attachment';
export const EMPLOYEE_URL = PAM_BASE_URL + '/query/employee/';
export const EMPLOYEE_NAME_URL = PAM_BASE_URL + '/query/employee-name/';
export const EMPLOYEE_VAX_HISTORY_URL = PAM_BASE_URL + '/query/vax-document/';
export const EMPLOYEE_TEST_HISTORY_URL = PAM_BASE_URL + '/query/test-result-document/';
export const VAX_TASK_DOCUMENT_URL = PAM_BASE_URL + '/query/task/document/';
export const DELETE_VAX_URL = PAM_BASE_URL + '/attestation/delete-vax-record/';
export const CONFIG_URL = PAM_BASE_URL + '/query/country-configs/';
export const SAVE_CONFIG_URL = PAM_BASE_URL + '/attestation/country-configs';

// List of countries allow to use this app
export const BLOCKED_COUNTRIES = window.REACT_APP_BLOCKED_COUNTRIES || []

export const KONAMI_CODE = [ "ArrowUp", "ArrowUp", "ArrowDown", "ArrowDown", "ArrowLeft", "ArrowRight", "ArrowLeft", "ArrowRight", "KeyB", "KeyA"];

// Maintenance Banner
export const MAINTENANCE_BANNER = window.REACT_APP_MAINTENANCE_BANNER || false
export const MAINTENANCE_BANNER_TEXT = window.REACT_APP_MAINTENANCE_BANNER_TEXT || ""

// Keycloak claims
export const KEYCLOAK_COUNTRY = window.REACT_APP_KEYCLOAK_COUNTRY
export const KEYCLOAK_EMAIL = window.REACT_APP_KEYCLOAK_EMAIL
export const KEYCLOAK_FIRST_NAME = window.REACT_APP_KEYCLOAK_FIRST_NAME
export const KEYCLOAK_ID = window.REACT_APP_KEYCLOAK_ID
export const KEYCLOAK_LAST_NAME = window.REACT_APP_KEYCLOAK_LAST_NAME
export const FEATURE_FLAGS = window.REACT_APP_FEATURE_FLAGS || []

// Maximum doses allowed by vaccine manufacturer
export const MAX_DOSES = window.REACT_APP_MAX_DOSES || {}

export const CONSENT_LABEL = window.REACT_APP_CONSENT_LABEL
export const ALTERNATE_CONSENT_LABEL = window.REACT_APP_ALTERNATE_CONSENT_LABEL

// Global country config
export const GLOBAL_CONFIG = window.REACT_APP_GLOBAL_CONFIG || {};

