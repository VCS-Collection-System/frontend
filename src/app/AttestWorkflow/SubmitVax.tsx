import React from "react";
import {
  Card,
  CardBody
} from "@patternfly/react-core";
import { useHistory } from "react-router-dom";
import { convertBase64ToBlob, convertBlobToBase64, dateConvert } from "@app/utils/utils";
import { useKeycloak } from "@react-keycloak/web";
import CardWithTitle from "@app/Components/CardWithTitle";
import CovidVaxForm from "@app/Components/CovidVaxForm";
import { getEmployeeInfo } from "@app/utils/getEmployeeInfo";

const SubmitVax: React.FunctionComponent = () => {
  const history = useHistory();
  const { keycloak } = useKeycloak();
  const employee = getEmployeeInfo(keycloak);
  const employeeFirstName = employee.firstName;
  const employeeLastName = employee.lastName;
  const employeeID = employee.id;

  const onLoad = () => {
    const data = sessionStorage.getItem("vax_image") || "";

    if (data !== "") {
      return new File([convertBase64ToBlob(data)], sessionStorage.getItem("vax_image_filename") || "");
    }

    return "";
  }

  const onSubmit = (data) => {
    sessionStorage.setItem("alt_first_name", data["alt_first_name"]);
    sessionStorage.setItem("alt_last_name", data["alt_last_name"]);
    sessionStorage.setItem("date_of_birth", dateConvert(data["date_of_birth"]));
    sessionStorage.setItem("vaccine", data["vaccine"]);
    sessionStorage.setItem("proof_type", data["proof_type"] || "OTHER");
    sessionStorage.setItem("vax_image_filename", data["vax_image"]?.name || "");
    for(let i = 1; typeof(data["vax_date_" + i]) !== "undefined"; i++) {
      sessionStorage.setItem("vax_date_" + i, dateConvert(data["vax_date_" + i]));
      sessionStorage.setItem("lot_number_" + i, data["lot_number_" + i]);
      sessionStorage.setItem("vaccine_" + i, data["vaccine_" + i]);
    }
    if(data["vax_image"]?.name) {
      convertBlobToBase64(data["vax_image"]).then((base64) => {
        sessionStorage["vax_image"] = base64;
      }).then(()=> history.push("/attest/review/vax"));
    } else {
      history.push("/attest/review/vax");
    }
  };

  return (
    <Card isRounded>
      <CardWithTitle title="Vaccination Form" info="Enter vaccination info" />
      <CardBody>
        <CovidVaxForm 
          isHRSubmit={false}
          employeeId={employeeID || ""}
          employeeName={(employeeFirstName || "") + " " + (employeeLastName || "")}
          onSubmit={onSubmit}
          onLoad={onLoad}
        />
      </CardBody>
    </Card>
  );
};

export { SubmitVax };
