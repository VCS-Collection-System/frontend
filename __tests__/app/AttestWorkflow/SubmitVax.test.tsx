import { act, fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { SubmitVax } from "../../../src/app/AttestWorkflow/SubmitVax"
import mockKeycloak from "../../../__mocks__/keycloakMock"
import { getConfig } from "../../../__mocks__/configMock"
import vaxImage from "../../../__mocks__/vaxImageMock"
import reactRouterDom from "react-router-dom"
import { convertBase64ToBlob, convertBlobToBase64, dateConvert } from "@app/utils/utils";
import { CONFIG_URL } from "@app/utils/constants"

jest.mock('@react-keycloak/web', () => ({
  useKeycloak: jest.fn( ()=>({keycloak: mockKeycloak}))
}))

jest.mock("@app/utils/constants", () => ({
  KEYCLOAK_FIRST_NAME: "firstName",
  KEYCLOAK_LAST_NAME: "lastName",
  KEYCLOAK_EMAIL: "email",
  KEYCLOAK_COUNTRY: "attributes.agency[0]",
  KEYCLOAK_ID: "attributes.workforceid[0]",
  MAX_DOSES: {"Moderna": 2},
  FEATURE_FLAGS: ["fakeFeature"],
  CONFIG_URL: "config/"
}))

const config = {
  "cze": {
    "proofOptional": true
  },
  "deu": {
    "proof": false
  },
  "default": {
    "allowPdf": false,
    "dateOfBirth": true,
    "consentCheckbox": false,
    "covidTest": false,
    "customInput": true,
    "hrdashboard": true,
    "hrsearch": true,
    "proof": true,
    "proofOptional": false,
    "cdc": true,
    "divoc": false,
    "greenPass": false,
    "recovery": false,
    "fakeFeature": true
  }
}

jest.mock('@app/utils/axiosInterceptor', () => ({
  get: (URL: string) => getConfig(URL, CONFIG_URL, config)
}))

jest.mock("react-router-dom")

jest.mock("@app/utils/utils", () => ({
  ...jest.requireActual("@app/utils/utils"),
  convertBlobToBase64: jest.fn( ()=> {return Promise.resolve({})})
}))

const mockPush = jest.fn()
reactRouterDom.useHistory = jest.fn().mockReturnValue({ push: mockPush })

describe('<SubmitVax />', ()=> {
  it('should be defined', () => {
    expect(SubmitVax).toBeDefined()
  })

  it("should match existing snapshot", async () => {
    expect(render(<SubmitVax />).asFragment()).toMatchSnapshot()
  })

  it("should not allow to submit without attachments", async () => {
    jest.spyOn(console,'error').mockReturnValue()
    await act(async () => {
      render(<SubmitVax />)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const acceptButton = screen.getByRole('button', { name: /Accept/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(mockPush).toBeCalledTimes(0); 
  })

  it("should be allowed to submit without attachments if optional is true", async () => {
    jest.spyOn(console,'error').mockReturnValue()
    mockKeycloak.profile = {
      firstName: "mockFirst",
      lastName: "mockLast",
      email: "mock@email.com",
      attributes: {
          workforceid: ["mockId"],
          agency: ["CZE"]
      }
    }
    await act(async () => {
      render(<SubmitVax />)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const acceptButton = screen.getByRole('button', { name: /Accept/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(mockPush).toHaveBeenCalledWith("/attest/review/vax");
  })

  it("should be allowed to submit without attachments if proof is false", async () => {
    jest.spyOn(console,'error').mockReturnValue()
    mockKeycloak.profile = {
      firstName: "mockFirst",
      lastName: "mockLast",
      email: "mock@email.com",
      attributes: {
          workforceid: ["mockId"],
          agency: ["DEU"]
      }
    }
    await act(async () => {
      render(<SubmitVax />)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const acceptButton = screen.getByRole('button', { name: /Accept/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(mockPush).toHaveBeenCalledWith("/attest/review/vax");
    expect(sessionStorage.getItem("proof_type")).toBe("OTHER");
  })

  it("should not display undefined when missing attributes", async () => {
    mockKeycloak.profile.firstName = "";
    mockKeycloak.profile.lastName = "";
    mockKeycloak.profile.attributes.workforceid = [];
    mockKeycloak.profile.attributes.agency=[];
    await act(async () => {
      render(<SubmitVax />)
    })

    expect(screen.queryByText("undefined")).toBe(null); 
  })

  it("should allow to submit one shot of complete form", async () => {
    sessionStorage.setItem("vax_image", "data:image/png;base64,VEhJUyBJUyBUSEUgQU5TV0VSCg")
    sessionStorage.setItem("vax_image_filename", "blob.pdf")
    await act(async () => {
      render(<SubmitVax />)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const acceptButton = screen.getByRole('button', { name: /Accept/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(mockPush).toHaveBeenCalledWith("/attest/review/vax")
  })

  it("should allow to submit two shots of complete form", async () => {
    sessionStorage.clear(); 
    sessionStorage.setItem("vax_image", "data:image/png;base64,VEhJUyBJUyBUSEUgQU5TV0VSCg"); 
    await act(async () => {
      render(<SubmitVax />)
    })

    const addDoseButton = screen.getByText('Add Dose') 
    await act(async () => {
      fireEvent.click(addDoseButton)
    })

    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect1 = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const covidBrandSelect2 = screen.getByRole('combobox', { name: /vaccine-vendor-2/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const acceptButton = screen.getByRole('button', { name: /Accept/i})

    await act(async () => {
      fireEvent.change(covidBrandSelect1, {target: {value: 'Moderna'}})
      fireEvent.change(covidBrandSelect2, {target: {value: 'Moderna'}})
    })

    const dateTwoInput = screen.getByRole('textbox', { name: /shot-2-date/i})
    const lotTwoInput = screen.getByRole('textbox', { name: /lot-number-2/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(dateTwoInput, {target: {value: '09-02-2021'}})
    fireEvent.change(lotTwoInput, {target: {value: '2'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(acceptButton)
    })

    expect(mockPush).toHaveBeenCalledWith("/attest/review/vax")
  })
})