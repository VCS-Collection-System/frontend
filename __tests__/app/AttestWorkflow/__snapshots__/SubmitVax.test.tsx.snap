// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitVax /> should match existing snapshot 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card pf-m-rounded"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <p
        class=""
        data-pf-content="true"
      >
        Vaccination Form
      </p>
      <div
        class="pf-c-content"
      >
        <small
          class=""
          data-pf-content="true"
        >
          Enter vaccination info
        </small>
      </div>
    </div>
    <div
      class="pf-c-card__body"
    >
      <form
        class="pf-c-form"
        novalidate=""
      >
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-label"
          >
            <label
              class="pf-c-form__label"
              for="employee-full-name"
            >
              <span
                class="pf-c-form__label-text"
              >
                Full Name
              </span>
            </label>
             
          </div>
          <div
            class="pf-c-form__group-control"
          >
            <input
              aria-invalid="false"
              aria-label="employee-full-name"
              class="pf-c-form-control"
              disabled=""
              type="text"
              value="mockFirst mockLast"
            />
            <div
              aria-live="polite"
              class="pf-c-form__helper-text"
              id="employee-full-name-helper"
            >
              Make sure this matches your full name.
            </div>
          </div>
        </div>
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-label"
          >
            <label
              class="pf-c-form__label"
              for="alt-first-name"
            >
              <span
                class="pf-c-form__label-text"
              >
                Preferred First Name (if applicable)
              </span>
            </label>
             
          </div>
          <div
            class="pf-c-form__group-control"
          >
            <input
              aria-invalid="false"
              aria-label="alt-first-name"
              class="pf-c-form-control"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              class="pf-c-form__helper-text"
              id="alt-first-name-helper"
            >
              Your first name as displayed on vaccination record if different than above.
            </div>
          </div>
        </div>
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-label"
          >
            <label
              class="pf-c-form__label"
              for="alt-last-name"
            >
              <span
                class="pf-c-form__label-text"
              >
                Preferred Last Name (if applicable)
              </span>
            </label>
             
          </div>
          <div
            class="pf-c-form__group-control"
          >
            <input
              aria-invalid="false"
              aria-label="alt-last-name"
              class="pf-c-form-control"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              class="pf-c-form__helper-text"
              id="alt-last-name-helper"
            >
              Your last name as displayed on vaccination record if different than above.
            </div>
          </div>
        </div>
        <div
          class="pf-l-grid pf-m-gutter"
        >
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-6-col-on-md"
          >
            <article
              class="pf-c-card"
              data-ouia-component-id="OUIA-Generated-Card-2"
              data-ouia-component-type="PF4/Card"
              data-ouia-safe="true"
              id=""
            >
              <div
                class="pf-c-card__title"
              >
                1st Dose or Booster 
              </div>
              <div
                class="pf-c-card__body"
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="shot-1-date"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Date
                      </span>
                      <span
                        aria-hidden="true"
                        class="pf-c-form__label-required"
                      >
                         *
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <div
                      class="pf-c-date-picker"
                      style="--pf-c-date-picker__input--c-form-control--width-chars: 10;"
                    >
                      <div
                        class="pf-c-date-picker__input"
                      >
                        <div
                          class="pf-c-input-group"
                        >
                          <input
                            aria-invalid="false"
                            aria-label="shot-1-date"
                            class="pf-c-form-control"
                            placeholder="MM-DD-YYYY"
                            type="text"
                            value=""
                          />
                          <button
                            aria-label="Toggle date picker"
                            class="pf-c-button pf-m-control"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 448 512"
                              width="1em"
                            >
                              <path
                                d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="shot-1-date-helper"
                    >
                      The date you received the dose.
                    </div>
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="vaccine-vendor-1"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Vaccine Manufacturer
                      </span>
                      <span
                        aria-hidden="true"
                        class="pf-c-form__label-required"
                      >
                         *
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <select
                      aria-invalid="false"
                      aria-label="vaccine-vendor-1"
                      class="pf-c-form-control"
                      data-ouia-component-id="OUIA-Generated-FormSelect-default-1"
                      data-ouia-component-type="PF4/FormSelect"
                      data-ouia-safe="true"
                    >
                      <option
                        class=""
                        disabled=""
                        value=""
                      >
                        Please Select
                      </option>
                      <option
                        class=""
                        value="AstraZeneca"
                      >
                        AstraZeneca
                      </option>
                      <option
                        class=""
                        value="Comirnaty"
                      >
                        Comirnaty
                      </option>
                      <option
                        class=""
                        value="Covaxin"
                      >
                        Covaxin
                      </option>
                      <option
                        class=""
                        value="Covishield"
                      >
                        Covishield
                      </option>
                      <option
                        class=""
                        value="Covovax"
                      >
                        Covovax
                      </option>
                      <option
                        class=""
                        value="Janssen"
                      >
                        Janssen
                      </option>
                      <option
                        class=""
                        value="Johnson"
                      >
                        Johnson & Johnson
                      </option>
                      <option
                        class=""
                        value="Moderna"
                      >
                        Moderna
                      </option>
                      <option
                        class=""
                        value="Novavax"
                      >
                        Novavax
                      </option>
                      <option
                        class=""
                        value="Oxford"
                      >
                        Oxford
                      </option>
                      <option
                        class=""
                        value="Pfizer"
                      >
                        Pfizer
                      </option>
                      <option
                        class=""
                        value="Sinopharm"
                      >
                        Sinopharm
                      </option>
                      <option
                        class=""
                        value="Sinovac_CoronaVac"
                      >
                        Sinovac-CoronaVac
                      </option>
                      <option
                        class=""
                        value="Spikevax"
                      >
                        Spikevax
                      </option>
                      <option
                        class=""
                        value="Sputnik"
                      >
                        Sputnik
                      </option>
                      <option
                        class=""
                        value="Vaxzevria"
                      >
                        Vaxzevria
                      </option>
                    </select>
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="vaccine-vendor-1-helper"
                    >
                      The vendor of this dose.
                    </div>
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="lot-number-1"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Lot/Batch Number
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="lot-number-1"
                      class="pf-c-form-control"
                      type="text"
                      value=""
                    />
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="lot-number-1-helper"
                    >
                      The lot/batch number of the vaccine that you received.
                    </div>
                  </div>
                </div>
              </div>
            </article>
          </div>
        </div>
        <a
          class=""
          data-pf-content="true"
          data-testid="add-dose"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
           Add Dose
        </a>
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-control"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-primary"
              data-ouia-component-id="OUIA-Generated-Button-primary-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="submit"
            >
              <span
                class="pf-c-button__icon pf-m-start"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  />
                </svg>
              </span>
              Accept
            </button>
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Back
            </button>
          </div>
        </div>
      </form>
    </div>
  </article>
</DocumentFragment>
`;
